opencti-platform/opencti-front/src/private/components/data/DataTableToolBar.jsx

```diff
@@ -648,6 +641,14 @@ class DataTableToolBar extends Component {
       removeIdAndIncorrectKeysFromFilterGroupObject(filters, availableFilterKeys),
     );
 
+    const selectedElementsList = R.values(selectedElements);
+    const { selectedTypes } = this.getSelectedTypes();
+    let scope = 'KNOWLEDGE';
+    if (selectedElementsList.some(({ entity_type }) => entity_type === 'Vocabulary')) {
+      scope = 'SETTINGS';
+    } else if (selectedTypes.at(0) === 'InternalFile' && !this.state.scope) {
+      scope = 'IMPORT';
+    }
```

Comment:
it's very bricolage... could we hide it under utility function `getBackgroundTaskScope(selectedList, selectedTypes)` and comment a bit the utility function ?

Reply:
Ok, I've explicited this extremely with a prop :)